File: bookstore-frontend/src/types/Book.ts
/**
 * Represents a single book as returned by the API.
 * Property names use camelCase to match the JSON serialization
 * that ASP.NET Core produces from the C# PascalCase model.
 */
export interface Book {
  bookID: number;         // Primary key from the database
  title: string;          // Full book title
  author: string;         // Author's full name
  publisher: string;      // Publishing company
  isbn: string;           // ISBN-13 identifier (e.g., "978-0451419439")
  classification: string; // Broad type: "Fiction" or "Non-Fiction"
  category: string;       // Sub-genre: "Biography", "Classic", "Historical", etc.
  pageCount: number;      // Total number of pages
  price: number;          // Retail price in USD
}

/**
 * Shape of the paginated response from GET /api/books.
 * The API wraps the book array with metadata so the frontend
 * can render pagination controls without a separate request.
 */
export interface BooksResponse {
  books: Book[];       // The books for the current page
  totalCount: number;  // Total books across all pages (within the active filter)
  totalPages: number;  // Total number of pages at the current page size
  currentPage: number; // Which page was returned
  pageSize: number;    // How many books were requested per page
}

/**
 * Represents a single line item in the shopping cart.
 * Extends Book with a quantity field.
 */
export interface CartItem {
  book: Book;      // The book that was added
  quantity: number; // How many copies are in the cart
}

/**
 * The state saved when navigating from the book list to the cart,
 * so the user can return to exactly where they left off.
 */
export interface BookListState {
  page: number;
  pageSize: number;
  sortOrder: 'asc' | 'desc';
  category: string; // Empty string means "All Categories"
}

File: BookstoreAPI/Models/Book.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookstoreAPI.Models
{
    /// <summary>
    /// Represents a single book in the bookstore database.
    /// Each property maps directly to a column in the SQLite "Books" table.
    /// The [Required] annotations ensure the model validator rejects any
    /// incomplete data before it ever reaches the database.
    /// </summary>
    [Table("Books")] // Explicitly maps this class to the "Books" table in SQLite
    public class Book
    {
        /// <summary>
        /// Primary key — auto-incremented by SQLite.
        /// The [Key] attribute tells Entity Framework this is the PK.
        /// </summary>
        [Key]
        public int BookID { get; set; }

        /// <summary>
        /// The full title of the book (e.g., "Les Miserables").
        /// </summary>
        [Required]
        public string Title { get; set; } = string.Empty;

        /// <summary>
        /// The name of the author (e.g., "Victor Hugo").
        /// </summary>
        [Required]
        public string Author { get; set; } = string.Empty;

        /// <summary>
        /// The publishing company (e.g., "Signet").
        /// </summary>
        [Required]
        public string Publisher { get; set; } = string.Empty;

        /// <summary>
        /// International Standard Book Number in ISBN-13 format
        /// (e.g., "978-0451419439"). Used as a globally unique book identifier.
        /// </summary>
        [Required]
        public string ISBN { get; set; } = string.Empty;

        /// <summary>
        /// Broad classification of the book's content
        /// (e.g., "Fiction" or "Non-Fiction").
        /// </summary>
        [Required]
        public string Classification { get; set; } = string.Empty;

        /// <summary>
        /// More specific genre or sub-category within the classification
        /// (e.g., "Biography", "Classic", "Historical").
        /// </summary>
        [Required]
        public string Category { get; set; } = string.Empty;

        /// <summary>
        /// Total number of pages in the book.
        /// </summary>
        [Required]
        public int PageCount { get; set; }

        /// <summary>
        /// Retail price in US dollars (e.g., 9.95).
        /// Uses decimal for accurate monetary arithmetic.
        /// </summary>
        [Required]
        public decimal Price { get; set; }
    }
}
